### PR TITLE
Fix namespace in Rebuild library

### PR DESCRIPTION
**Pull request by RikkiGibson**
cc @jaredpar @cston for a small review

This has to go in after #52208 to reduce conflicts.

### CODE REVIEW

File: Roslyn.sln [hunk 1]
```diff
@@ -483,7 +483,7 @@ Project("{D954291E-2A0B-460D-934E-DC6B0785DB48}") = "Microsoft.CodeAnalysis.Coll
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.Collections.Package", "src\Dependencies\Collections\Microsoft.CodeAnalysis.Collections.Package.csproj", "{0C2E1633-1462-4712-88F4-A0C945BAD3A8}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Rebuild", "src\Compilers\Core\Rebuild\Rebuild.csproj", "{B7D29559-4360-434A-B9B9-2C0612287999}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Rebuild", "src\Compilers\Core\Rebuild\Microsoft.CodeAnalysis.Rebuild.csproj", "{B7D29559-4360-434A-B9B9-2C0612287999}"
```
Comment (Youssef1313) [inline Roslyn.sln:486]:
```suggestion
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.Rebuild", "src\Compilers\Core\Rebuild\Microsoft.CodeAnalysis.Rebuild.csproj", "{B7D29559-4360-434A-B9B9-2C0612287999}"
```

File: Compilers.sln [hunk 2]
```diff
@@ -168,7 +168,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CodeStyleConfigFileGenerato
 EndProject
 Project("{D954291E-2A0B-460D-934E-DC6B0785DB48}") = "Microsoft.CodeAnalysis.Collections", "src\Dependencies\Collections\Microsoft.CodeAnalysis.Collections.shproj", "{E919DD77-34F8-4F57-8058-4D3FF4C2B241}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Rebuild", "src\Compilers\Core\Rebuild\Rebuild.csproj", "{321F9FED-AACC-42CB-93E5-541D79E099E8}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Rebuild", "src\Compilers\Core\Rebuild\Microsoft.CodeAnalysis.Rebuild.csproj", "{321F9FED-AACC-42CB-93E5-541D79E099E8}"
```
Comment (Youssef1313) [inline Compilers.sln:171]:
```suggestion
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.Rebuild", "src\Compilers\Core\Rebuild\Microsoft.CodeAnalysis.Rebuild.csproj", "{321F9FED-AACC-42CB-93E5-541D79E099E8}"
```

File: src/Tools/BuildValidator/BuildValidator.sln [hunk 3]
```diff
@@ -5,7 +5,7 @@ VisualStudioVersion = 16.0.30905.15
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "BuildValidator", "BuildValidator.csproj", "{F3EDD192-7E74-441D-B096-16C0FE2731E3}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Rebuild", "..\..\Compilers\Core\Rebuild\Rebuild.csproj", "{04CCBB7F-B1E3-4E76-A4A7-C648EBE77CBB}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Rebuild", "..\..\Compilers\Core\Rebuild\Microsoft.CodeAnalysis.Rebuild.csproj", "{04CCBB7F-B1E3-4E76-A4A7-C648EBE77CBB}"
```
Comment (Youssef1313) [inline src/Tools/BuildValidator/BuildValidator.sln:8]:
```suggestion
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.Rebuild", "..\..\Compilers\Core\Rebuild\Microsoft.CodeAnalysis.Rebuild.csproj", "{04CCBB7F-B1E3-4E76-A4A7-C648EBE77CBB}"
```

**Comment by RikkiGibson** [comment]
Thanks @Youssef1313!

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/Core/RebuildTest/RoundTripUtil.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@
 using System.Reflection;
 using System.Reflection.PortableExecutable;
 using System.Threading;
-using BuildValidator;
+using Microsoft.CodeAnalysis.Rebuild;
```
Comment (RikkiGibson) [inline src/Compilers/Core/RebuildTest/RoundTripUtil.cs:12]: This using is not sorted. Will clean it up in a subsequent PR.

File: src/Compilers/Core/RebuildTest/OptionRoundTripTests.cs [hunk 5]
```diff
@@ -10,12 +10,12 @@
...
 using System.Text;
 using System.Threading;
-using BuildValidator;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Emit;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Rebuild;
 using Microsoft.CodeAnalysis.VisualBasic;
```
Comment (Youssef1313) [inline src/Compilers/Core/RebuildTest/OptionRoundTripTests.cs:19]: nit: swap to keep ordered alphabetically.

```suggestion
using Microsoft.CodeAnalysis.Rebuild;
using Microsoft.CodeAnalysis.Test.Utilities;
using Microsoft.CodeAnalysis.VisualBasic;
```

File: src/Compilers/Core/RebuildTest/RoundTripUtil.cs (hunk 4 above)
Comment (Youssef1313) [inline src/Compilers/Core/RebuildTest/RoundTripUtil.cs:12]: nit: after  `Microsoft.CodeAnalysis.Emit` to keep ordered alphabetically.

**Review by cston — APPROVED** [review]

**Pull request merged**